client.go

```diff
@@ -396,3 +409,38 @@ func (c *Client) httpRequest(
 func (c *Client) requestPath(path string) string {
 	return fmt.Sprintf("%s:%v/%s", c.hostname, c.port, path)
 }
+
+type authRequest struct {
+	Namespace string `json:"namespace"`
+	Password  string `json:"password"`
+}
+
+type authResponse struct {
+	Token string `json:"access_token"`
```

Comment:
Although now I am wondering if the term token is confusing given the naming of this? Thoughts?

Reply:
I should have kept reading. yes confusion. I have also seen merge #83 with the rename, but still think there will be confusion.

how about:
* leave the word token for the JWT auth token
* use Key for the "password" - it is the key to generating tokens
* label each API Key with a "Username"

As discussed, the username can be used to specify a key for revocation. The username also allows for usage audit - "who might use that key?"

Reply:
So I ended up calling the username "unique_name" to try and encourage people to do user / device naming. If we went with the key name, a namespace would be composed of a series of keys, each of which has a unique name. I think that's reasonable. Agreed?

Reply:
I think unique_name is a bit ugly and non-specific. unique to?
maybe username or key_name

I assume auth toekn requests will include key_name?

Reply:
So, to create a new namespace (or add a key to a namespace) would hand over: namespace, key_name, and key. To auth you just hand over namespace and key and we ignore the key_name.

Reply:
Since the users choose their own key there is the possibility of key collision.
Auth will require the key_name.
We might want to generate the keys to ensure secure keys. Either way, key_name should be specified.

Reply:
I guess key collision in the same namespace does not matter. Unless you running an audit log.

Reply:
I guess the same is true with a key name too though right? I could use "username" and "password", and if you do the same we still collide. We could return an error when creating the second key entry, but that would also signal that you'd managed to collide and therefore couldn't be auditted. I'm not entirely sure of a good solution here.